lib/presentation/ui/occurrences_time_widget.dart

```diff
@@ -0,0 +1,122 @@
+                                ),
+                                new Text(
+                                  this._startTime != null ? this._startTime.day.toString() + "-" + this._startTime.month.toString() + "-" + this._startTime.year.toString() : "--",
+                                  overflow: TextOverflow.clip,
+                                  style: DefaultTextStyle.of(context).style.apply(
+                                      fontSizeFactor: 0.5,
+                                      decoration: TextDecoration.none,
+                                      color: Colors.lightGreen
+                                  ),
+                                )
+                              ],
+                            ),
+                            new Column(
+                              mainAxisSize: MainAxisSize.min,
+                              children: <Widget>[
+                                new Text(
+                                  "Fim",
+                                  overflow: TextOverflow.clip,
+                                  style: DefaultTextStyle.of(context).style.apply(
+                                      fontSizeFactor: 0.5,
+                                      decoration: TextDecoration.none,
+                                      color: Colors.grey
+                                  ),
+                                ),
+                                new Text(
+                                  this._endTime != null ? this._endTime.hour.toString() + ":" + this._endTime.minute.toString() : "--",
+                                  overflow: TextOverflow.clip,
+                                  style: DefaultTextStyle.of(context).style.apply(
+                                      fontSizeFactor: 0.5,
+                                      decoration: TextDecoration.none,
+                                      color: Colors.lightGreen
+                                  ),
+                                ),
+                                new Text(
+                                  this._endTime != null ? this._endTime.day.toString() + "-" + this._endTime.month.toString() + "-" + this._endTime.year.toString() : "--",
+                                  overflow: TextOverflow.clip,
+                                  style: DefaultTextStyle.of(context).style.apply(
+                                      fontSizeFactor: 0.5,
+                                      decoration: TextDecoration.none,
+                                      color: Colors.lightGreen),
+                                )
+                              ],
+                            ),
+                          ],
+                        ),
+                        new Row(
+                          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                          children: <Widget>[
+                            new Text(
+                        this._lastUpdated != null ?
```

Comment:
As with the previous statement, if you could extract a method and use date formatting it would be awesome.

This way the widget tree would not have this kind of logic

Reply:
@Vanethos , please look at my updated PR